### PR TITLE
Add filter section side view

### PR DESCRIPTION
**Pull request by brianmugweru**
Make sure the checklist below is checked before PR merge, mark if not applicable

## PR Checklist

- [ ] UI created matched design
- [ ] For evey color class added, dark: is defined, images have alt and \_blank links have rel
- [ ] API is integrated as required
- [ ] Page is responsive

## What is the Purpose?
Filter side section view user interface 

## What was the approach?
- Added the UI component for filter section
- Added functionality app.tsx to view and close filter section
- Setup filter section to filter between the different section offered

## Are there any concerns to addressed further before or after merging this PR?
- Later on we'll have to improve and add API's fetches on filters to it

## Mentions?
@manuelgeek @cliffgor

## Issue(s) affected?
https://github.com/droidconKE/droidconKE2022Web/issues/34

## Screenshots
Screenshot 2022-09-20 at 13 52 40

### CODE REVIEW

File: components/sessions/FilterSessions.tsx [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    setFilter((prev: FilterInterface) => ({
+      ...prev,
+      [name]: value,
+    }))
+  }
+
+  return (
+    <div className="fixed right-0 top-0 w-25 bg-black z-20 text-white lg:w-1/3 md:w-2/3 h-screen transition-all ease-in-out duration-1000 transform translate-x-0">
+      <div className="bg-black relative">
+        <div className="py-16 px-10">
```
Comment (manuelgeek) [inline components/sessions/FilterSessions.tsx:51]: px-4 md:px-10 ?

let's keep it at 4 in mobile

File: components/sessions/FilterSessions.tsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
+import { useState, MouseEvent } from 'react'
+import type { NextPage } from 'next'
+import { groupBy3 } from '../../utils/helpers'
+
+interface FilterSessionProps {
```
Comment (manuelgeek) [inline components/sessions/FilterSessions.tsx:5]: could you move these interfaces to `types/types.ts`

File: components/sessions/FilterSessions.tsx [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            ))}
+          </div>
+
+          <div className="mt-10">
+            <div className="flex pb-1">
+              <button
+                className="w-full rounded-md py-2 bg-primary font-bold"
+                type="button"
+              >
+                FILTER
```
Comment (manuelgeek) [inline components/sessions/FilterSessions.tsx:256]: change to lowercase

Comment (manuelgeek) [reply components/sessions/FilterSessions.tsx:256]: also there's a styling fo button
btn-primary

`className="btn-primary w-full"`

**Comment by manuelgeek** [comment]
@brianmugweru also fix conflicts

**Review by manuelgeek — APPROVED** [review]

**Review by i-Krypt — APPROVED** [review]

**Pull request merged**